tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,144 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
+// 1. create a user created vmgroup "vmgroup_test1" and add VMs to this vmgroup
+// 2. create two volumes for this vmgroup, one is on default datastore, and the other is on non default_datastore
+// 3. verify mount/unmount for those two volumes
+// 4. after unmount, verify volume is detached for those two volumes
+// 5. remove VMs from this vmgroup, and remove the vmgroup with "--remove-volume" option
+// 6. verify that volumes do not exist after removing vmgroup
+func (vg *VmGroupTest) TestVmgroupRemoveWithRemoveVol(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume in vgTestVMgroup1 on default_datastore
+	vg.createVolumes(c, vg.volName1)
+
+	// Add create privilege for the second datastore
+	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create another volume on second datastore
+	// need to use full name for volume name
+	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+	// Verify volume can be mounted and unmounted for the first volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.Datastores[0], vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+	// Verify volume can be mounted and unmounted for the second volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status = verification.VerifyDetachedStatus(vg.volName2, vg.config.Datastores[1],
+		vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName2+"@"+vg.config.Datastores[1]))
+
+	// remove VM1 and VM2 from vgTestVMgroup1 and then remove the vmgroup
+	vmList := []string{vg.config.DockerHostNames[0], vg.config.DockerHostNames[1]}
+	for _, vm := range vmList {
+		out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vm)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+
+	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, true)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// vmgroup has been removed and all volumes belong to it have been removed
+	// "err.Error()" will be filled with "exit status 1"
+	volumeList := []string{vg.volName1, vg.volName2}
+	for _, volume := range volumeList {
+		out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, volume)
+		log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, volume)
+		c.Assert(err.Error(), Equals, "exit status 1", Commentf("volume %s should be removed", volume))
+	}
+
+	// Recreate the test VM group1 and add VM1, VM2 back to the vmgroup
```

Comment:
Why? I checked the existing tests in "vmgroups_test.go", and I followed the approach as other tests. Why it will lead to confusion? Comments already say what is the purpose of the code from L494-502.